packages/kamino-sdk/src/Kamino.ts

```diff
@@ -2858,6 +2860,122 @@ export class Kamino {
     return initializeStrategy(strategyArgs, strategyAccounts);
   };
 
+  /**
+   * Get transaction instruction to close Kamino strategy, including its position if there is any
+   * and strategy token accounts.
+   * @param strategy public key of the strategy
+   * @returns instruction to close the strategy
+   */
+  closeStrategy = async (strategy: PublicKey) => {
+    const { address: _strategyPubkey, strategy: strategyState } = await this.getStrategyStateIfNotFetched(strategy);
+    let collInfos = await this.getCollateralInfos();
+
+    const poolProgram = getDexProgramId(strategyState);
+    let oldPositionOrBaseVaultAuthority = strategyState.baseVaultAuthority;
+    let oldPositionMintOrBaseVaultAuthority = strategyState.baseVaultAuthority;
+    let oldPositionTokenAccountOrBaseVaultAuthority = strategyState.baseVaultAuthority;
+    if (!strategyState.position.equals(DEFAULT_PUBKEY)) {
+      oldPositionOrBaseVaultAuthority = strategyState.position;
+      oldPositionMintOrBaseVaultAuthority = strategyState.positionMint;
+      oldPositionTokenAccountOrBaseVaultAuthority = strategyState.positionTokenAccount;
+    }
+    let userTokenAAta = getAssociatedTokenAddress(strategyState.tokenAMint, strategyState.adminAuthority);
+    let userTokenBAta = getAssociatedTokenAddress(strategyState.tokenBMint, strategyState.adminAuthority);
+    let reward0Vault = strategyState.baseVaultAuthority;
+    let userReward0Ata = strategyState.baseVaultAuthority;
+    if (!strategyState.reward0Vault.equals(DEFAULT_PUBKEY) && strategyState.reward0Decimals.toNumber() > 0) {
```

Comment:
can you make an util where you pass the vault and decimals and does this check or maybe even returns directly the ATA you need? it is very duplicated

Reply:
refactored the condition in a separate func